pkg/collector/check/logreceiver.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package check
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+)
+
+// checkContext holds a list of reference to different components used by Go and Python checks.
+//
+// This is a temporary solution until checks are components themselves and can request dependencies through FX.
+//
+// This also allows Go function exported to CPython to recover there reference to different components when coming out
+// of C to Go. This way python checks can submit metadata to inventorychecks through the 'SetCheckMetadata' python
+// method.
+type logReceiver struct {
+	lr integrations.Component
+}
+
+var logRcv logReceiver
```

Comment:
you can probably just hold a ref to the component here instead of wrapping it in `logReceiver` (since that struct has no other properties)

Reply:
Makes sense 👍